fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadJob.java

```diff
@@ -1532,11 +1532,11 @@ public List<String> getShowInfo() {
             row.add(otherMsg);
 
             if (RunMode.getCurrentRunMode() == RunMode.SHARED_DATA) {
-                Warehouse warehouse = GlobalStateMgr.getCurrentState().getWarehouseMgr().getWarehouse(warehouseId);
-                if (warehouse != null) {
+                try {
+                    Warehouse warehouse = GlobalStateMgr.getCurrentState().getWarehouseMgr().getWarehouse(warehouseId);
                     row.add(warehouse.getName());
-                } else {
-                    row.add("");
+                } catch (Exception e) {
+                    row.add(e.getMessage());
```

Comment:
I suggest the error message could be `warehouse 'id:$id' does not exist`, it's clearer.